File: djangoBackend/update-service/updateserv/utils/read_service.py
import httpx
import json
from django.conf import settings

# This must be the correct URL, with NO space at the end of the string.
READ_SERVICE_URL = "http://localhost:4000/read/api/get-data" 
#                                         ^ NO SPACE HERE

async def fetch_record(table, id, include_password, user, token):
    try:
        async with httpx.AsyncClient() as client:
            response = await client.get(
                READ_SERVICE_URL,
                params={"tableName": table, "id": id, "includePassword": include_password},
                headers={
                    "x-user": json.dumps(user),
                    "Authorization": token
                }
            )
        data = response.json()
        return data.get("data", [])
    except Exception as e:
        print("READ-SERVICE ERROR:", e)
        return None

File: backend/update-service/utils/readService.js
// updateservice/utils/readService
import axios from "axios";

const READ_SERVICE_URL = process.env.READ_SERVICE_URL;

export async function fetchRecord(table, id, includePassword, user, token) {
  try {
    const response = await axios.get(READ_SERVICE_URL, {
      params: { tableName: table, id, includePassword },
      headers: {
        "x-user": JSON.stringify(user),
        Authorization: token,
      },
    });

    // gateway already decrypted
    return response.data.data || [];
  } catch (err) {
    console.error("READ-SERVICE ERROR:", err?.response?.data || err);
    return null;
  }
}
